client: Invalid memory reference during Firo trade.

I noticed this after leaving Basic MM for Firo-USDC running overnight. 

<details>

<summary>Log + Stack trace</summary>

```
2025-04-17 03:56:07.922 [DBG] CORE: Received valid ack for 'init' request for match 83f38dfce055603c0ff1c1469d04e361398b1915171d614ea8f4a3ad08a5304a
2025-04-17 03:56:10.752 [INF] CORE: Contract for match 83f38dfce055603c0ff1c1469d04e361398b1915171d614ea8f4a3ad08a5304a with swap coin 0xed775101a5d7d8463b4627affa2373b132ec4f694912b4868659c2b3d3cf8b6d (USDC.POLYGON) expires at 2025-04-17 19:49:45 +0800 +08 (7h53m34s).
2025-04-17 03:56:11.017 [DBG] CORE: Slow tick: trade 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 processed in 796.376797ms, blocked for 352ns
2025-04-17 03:56:25.009 [DBG] CORE: Slow tick: trade 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 processed in 797.830236ms, blocked for 140ns
2025-04-17 03:56:40.751 [DBG] CORE: Slow tick: trade 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 processed in 796.201788ms, blocked for 208ns
2025-04-17 03:57:30.626 [DBG] CORE: Slow tick: trade 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 processed in 572.025588ms, blocked for 218ns
2025-04-17 03:57:34.716 [DBG] CORE: Slow tick: trade 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 processed in 622.999871ms, blocked for 258ns
2025-04-17 03:57:40.774 [ERR] CORE: HTTP error: "404 Not Found" (code 404)
2025-04-17 03:57:44.689 [DBG] CORE: Slow tick: trade 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 processed in 610.987115ms, blocked for 322ns
2025-04-17 03:58:29.872 [INF] CORE: Notified of maker's redemption (usdc.polygon: 0x0c18c3da801fc9c9d044f75a0bf8923a753b24a47e89c59d56a68a5892392867) and validated secret for order 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978...
2025-04-17 03:58:29.877 [DBG] CORE: Redeemable match 83f38dfce055603c0ff1c1469d04e361398b1915171d614ea8f4a3ad08a5304a for order 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978 (Taker)
2025-04-17 03:58:30.118 [INF] CORE: Broadcasted redeem transaction spending 1 contracts for order 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978, paying to aef746fa3b8b7208d06aea78281910010f5fce9bc2647bdf89215786c1037f79:0 (firo)
2025-04-17 03:58:30.123 [INF] CORE: Match 83f38dfce055603c0ff1c1469d04e361398b1915171d614ea8f4a3ad08a5304a complete: buy 100000000 firo
2025-04-17 03:58:30.126 [DBG] CORE: Notifying DEX dex.decred.org:7232 of our firo swap redemption aef746fa3b8b7208d06aea78281910010f5fce9bc2647bdf89215786c1037f79:0 for match 83f38dfce055603c0ff1c1469d04e361398b1915171d614ea8f4a3ad08a5304a
2025-04-17 03:58:30.126 [DBG] CORE: notify: |POKE| (order) Match complete - Redeemed 1.00000000 FIRO on order {{{order|6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978}}} - Order: 6a0b64e3d3f80a57e5864f481594e7d1d58c6facca1941644325065499bce978

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12177e2]

goroutine 4093 [running]:
decred.org/dcrdex/client/asset/btc.(*BadgerTxDB).GetTx(0x0, {0xc000fee580, 0x40})
	/home/dev/dex/dcrdex/client/asset/btc/txdb.go:394 +0x42
decred.org/dcrdex/client/asset/btc.(*ExchangeWalletElectrum).WalletTransaction(0xc000788a50, {0x1b97860?, 0xc000794000?}, {0xc01182aff0, 0x48})
	/home/dev/dex/dcrdex/client/asset/btc/electrum.go:550 +0x13c
decred.org/dcrdex/client/core.(*xcWallet).WalletTransaction(0xc000794000, {0x265ce30, 0xc0002aeaa0}, {0xc01182aff0, 0x48})
	/home/dev/dex/dcrdex/client/core/wallet.go:583 +0x8e
decred.org/dcrdex/client/core.(*Core).WalletTransaction(0xc0005a6c00, 0x88, {0xc01182aff0, 0x48})
	/home/dev/dex/dcrdex/client/core/core.go:5817 +0xf0
decred.org/dcrdex/client/mm.(*pendingDEXOrder).updateState.func1(...)
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2704
decred.org/dcrdex/client/mm.(*pendingDEXOrder).updateState(0xc02ee87480, 0xc00cb0cd80, 0xc006c15d90, 0x1084a, 0x56208)
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2714 +0x4f5
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).handleDEXOrderUpdate(0xc01041e400, 0xc00cb0cd80)
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2741 +0x199
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).handleDEXNotification(0xc01041e400, {0x2668fa0?, 0xc02953b100?})
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2815 +0x545
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect.func3()
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:3427 +0xc5
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect in goroutine 3796
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:3417 +0x35b

```

</details>

<details>

<summary>Lead up to invalid memory address ..  </summary>

```
2025-04-17 07:49:24.124 [INF] CORE: Match 596ab69a52e622a89501d6a0bac44ca25705b64dca160a628ebd8fa7b960c62b not yet redeemable: current confs = 0, required confs = 2
2025-04-17 07:49:24.155 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 580.911504ms, blocked for 342ns
2025-04-17 07:49:30.146 [DBG] CORE: Received preimage request for order b5e2ea7736e27d26d398e0a73d5eb8562c5ceb64f3f0dc73bb67c31298912a6a with known commitment 1d90e64ab08018c80e3f60b776fc8f3bc41eb02ee01ef1a90e486e3bcb765b81
2025-04-17 07:49:30.146 [DBG] CORE: Received preimage request for order 3bd71252b318d036829afd2ac476d42af16f59d3b53490c51ef6ce736509207d with known commitment 4bdc00c6d41f6d1de55ececbe900156d7da699db080c77f93f04764801ac71ff
2025-04-17 07:49:30.147 [DBG] CORE: async processPreimageRequest for b5e2ea7736e27d26d398e0a73d5eb8562c5ceb64f3f0dc73bb67c31298912a6a succeeded
2025-04-17 07:49:30.149 [DBG] CORE: Received preimage request for order 9c5a3b00eb6bdb0ff14dcd77b9e5a51981c0fee578fa1d447b55811042873d9c with known commitment 9028fa1b466cadea42fd650ca945fae898056f8ed23b16791939a70260b113fb
2025-04-17 07:49:30.149 [DBG] CORE: Received preimage request for order c9ccb480f3c1e6c7dc1dac60420ffc3cf706d854124f5b213785b6cda019a9ff with known commitment 9caff48a69a18e9f4c82d3fd65df68ee10922bb4955d2669797cc870fc4aef3c
2025-04-17 07:49:30.149 [DBG] CORE: Received preimage request for order c7b9e642b1936c6cfa5c3da065efd18330a7097afbef701b8eb5244f22f405b4 with known commitment c1b0be14de5a1c24a3ee7d8af2a93dd8c58f86c6a05aee48bfbaf5dc6b33564b
2025-04-17 07:49:30.149 [DBG] CORE: Received preimage request for order ea8ac458602b797670108242d0adc82569491cbb88c3509f0c5061da3540a64f with known commitment e14647d0faa30a3fff8092747ca751eeb5e06960bc38fd9d2205034d7b967900
2025-04-17 07:49:30.149 [DBG] CORE: Received preimage request for order 271ab5ed080be0f838f57baf360bd3934a9ef340497466f22e3bfc56ea7fbeb2 with known commitment fed27c9d3dabbffb7efdb0d397d3a38b862b62c7c9fae3645e45352626b508fe
2025-04-17 07:49:30.150 [DBG] CORE: async processPreimageRequest for 9c5a3b00eb6bdb0ff14dcd77b9e5a51981c0fee578fa1d447b55811042873d9c succeeded
2025-04-17 07:49:30.150 [DBG] CORE: async processPreimageRequest for c7b9e642b1936c6cfa5c3da065efd18330a7097afbef701b8eb5244f22f405b4 succeeded
2025-04-17 07:49:30.150 [DBG] CORE: async processPreimageRequest for ea8ac458602b797670108242d0adc82569491cbb88c3509f0c5061da3540a64f succeeded
2025-04-17 07:49:30.150 [DBG] CORE: async processPreimageRequest for 271ab5ed080be0f838f57baf360bd3934a9ef340497466f22e3bfc56ea7fbeb2 succeeded
2025-04-17 07:49:30.150 [DBG] CORE: async processPreimageRequest for 3bd71252b318d036829afd2ac476d42af16f59d3b53490c51ef6ce736509207d succeeded
2025-04-17 07:49:30.150 [DBG] CORE: async processPreimageRequest for c9ccb480f3c1e6c7dc1dac60420ffc3cf706d854124f5b213785b6cda019a9ff succeeded
2025-04-17 07:49:30.795 [DBG] CORE: Registering match beecbb210f0466152333af6d33765901b037b61884a75ab411e9e7ca9e8aad12 for order a8db5e7eac5d133a86e4b0ba46898157accd82caccdc3d13ed6bb35edb30b3ba (Maker) in status NewlyMatched
2025-04-17 07:49:30.796 [DBG] CORE: Registering match beecbb210f0466152333af6d33765901b037b61884a75ab411e9e7ca9e8aad12 for order 9c5a3b00eb6bdb0ff14dcd77b9e5a51981c0fee578fa1d447b55811042873d9c (Taker) in status NewlyMatched
2025-04-17 07:49:30.798 [DBG] CORE: Registering match 0b3ce09a1c8bbd05d06f7c8e9aa0b3e46ef07475f02ae0a44edb242c8b0d3a64 for order 48903bcae33988bfcc46e25acd4102970d24f6833dc8acff89b8747f67366fe8 (Maker) in status NewlyMatched
2025-04-17 07:49:30.799 [DBG] CORE: Registering match 0b3ce09a1c8bbd05d06f7c8e9aa0b3e46ef07475f02ae0a44edb242c8b0d3a64 for order c9ccb480f3c1e6c7dc1dac60420ffc3cf706d854124f5b213785b6cda019a9ff (Taker) in status NewlyMatched
2025-04-17 07:49:30.799 [DBG] CORE: Registering match 9933d3d0dfc4024623437e821e9f2194c68887358c3939d17cf319f0c1d4bcd1 for order c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 (Maker) in status NewlyMatched
2025-04-17 07:49:30.799 [DBG] CORE: Registering match 9933d3d0dfc4024623437e821e9f2194c68887358c3939d17cf319f0c1d4bcd1 for order ea8ac458602b797670108242d0adc82569491cbb88c3509f0c5061da3540a64f (Taker) in status NewlyMatched
2025-04-17 07:49:30.799 [DBG] CORE: Registering match 013d2e7df9e7f181f2b11d65686b28c36358624dbb6ec701e7c484f2f086f376 for order ca85496ab036a19b998f3b4535339d2519d4b3dc59747c9ee1c327c0543d77a1 (Maker) in status NewlyMatched
2025-04-17 07:49:30.799 [DBG] CORE: Registering match 013d2e7df9e7f181f2b11d65686b28c36358624dbb6ec701e7c484f2f086f376 for order c7b9e642b1936c6cfa5c3da065efd18330a7097afbef701b8eb5244f22f405b4 (Taker) in status NewlyMatched
2025-04-17 07:49:30.799 [DBG] CORE: Registering match 39fe766de5c115001a5fc50f121a8e317f14b569ede6a3af3c4864a92af6d33c for order 18320727da8c65973a81c6f65c6f158a97e63e3d7b60d0758d15587240d6343c (Maker) in status NewlyMatched
2025-04-17 07:49:30.800 [DBG] CORE: Registering match 39fe766de5c115001a5fc50f121a8e317f14b569ede6a3af3c4864a92af6d33c for order 271ab5ed080be0f838f57baf360bd3934a9ef340497466f22e3bfc56ea7fbeb2 (Taker) in status NewlyMatched
2025-04-17 07:49:30.800 [DBG] CORE: Registering match 0caa1bff66871c3f7453405a35378982b2b3c16ef4160282ea1c7f8fbda25422 for order 30b58df021688fa23afb97d35fa756890fd60f999080bb61fb705f4c9edddd41 (Maker) in status NewlyMatched
2025-04-17 07:49:30.800 [DBG] CORE: Registering match 0caa1bff66871c3f7453405a35378982b2b3c16ef4160282ea1c7f8fbda25422 for order b5e2ea7736e27d26d398e0a73d5eb8562c5ceb64f3f0dc73bb67c31298912a6a (Taker) in status NewlyMatched
2025-04-17 07:49:30.800 [DBG] CORE: Registering match f323c46141d72da9387eccf039a81c9524ebcf20b043235e0a27f5b541a95380 for order 14d4f6f7134c12facb006b6dbc300f8ba937df0d1c3b5347495816c9a4ed2e19 (Maker) in status NewlyMatched
2025-04-17 07:49:30.800 [DBG] CORE: Registering match f323c46141d72da9387eccf039a81c9524ebcf20b043235e0a27f5b541a95380 for order 3bd71252b318d036829afd2ac476d42af16f59d3b53490c51ef6ce736509207d (Taker) in status NewlyMatched
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order c7b9e642b1936c6cfa5c3da065efd18330a7097afbef701b8eb5244f22f405b4 received. Match id = 013d2e7df9e7f181f2b11d65686b28c36358624dbb6ec701e7c484f2f086f376
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order 3bd71252b318d036829afd2ac476d42af16f59d3b53490c51ef6ce736509207d received. Match id = f323c46141d72da9387eccf039a81c9524ebcf20b043235e0a27f5b541a95380
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order 271ab5ed080be0f838f57baf360bd3934a9ef340497466f22e3bfc56ea7fbeb2 received. Match id = 39fe766de5c115001a5fc50f121a8e317f14b569ede6a3af3c4864a92af6d33c
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order b5e2ea7736e27d26d398e0a73d5eb8562c5ceb64f3f0dc73bb67c31298912a6a received. Match id = 0caa1bff66871c3f7453405a35378982b2b3c16ef4160282ea1c7f8fbda25422
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order ea8ac458602b797670108242d0adc82569491cbb88c3509f0c5061da3540a64f received. Match id = 9933d3d0dfc4024623437e821e9f2194c68887358c3939d17cf319f0c1d4bcd1
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order c9ccb480f3c1e6c7dc1dac60420ffc3cf706d854124f5b213785b6cda019a9ff received. Match id = 0b3ce09a1c8bbd05d06f7c8e9aa0b3e46ef07475f02ae0a44edb242c8b0d3a64
2025-04-17 07:49:30.800 [DBG] CORE: Taker notification for cancel order 9c5a3b00eb6bdb0ff14dcd77b9e5a51981c0fee578fa1d447b55811042873d9c received. Match id = beecbb210f0466152333af6d33765901b037b61884a75ab411e9e7ca9e8aad12
2025-04-17 07:49:30.805 [INF] CORE: Order ca85496ab036a19b998f3b4535339d2519d4b3dc59747c9ee1c327c0543d77a1 canceled. match id = 013d2e7df9e7f181f2b11d65686b28c36358624dbb6ec701e7c484f2f086f376
2025-04-17 07:49:30.805 [DBG] CORE[firo]: returning coins [d140fd7d3f4a0d1520cc0690b136710b283fd32bfd3137b57acfca5ccb3eb057:1]
2025-04-17 07:49:30.808 [INF] CORE: Order 14d4f6f7134c12facb006b6dbc300f8ba937df0d1c3b5347495816c9a4ed2e19 canceled. match id = f323c46141d72da9387eccf039a81c9524ebcf20b043235e0a27f5b541a95380
2025-04-17 07:49:30.808 [DBG] CORE[firo]: returning coins [d140fd7d3f4a0d1520cc0690b136710b283fd32bfd3137b57acfca5ccb3eb057:2]
2025-04-17 07:49:30.811 [INF] CORE: Order 18320727da8c65973a81c6f65c6f158a97e63e3d7b60d0758d15587240d6343c canceled. match id = 39fe766de5c115001a5fc50f121a8e317f14b569ede6a3af3c4864a92af6d33c
2025-04-17 07:49:30.814 [INF] CORE: Order 30b58df021688fa23afb97d35fa756890fd60f999080bb61fb705f4c9edddd41 canceled. match id = 0caa1bff66871c3f7453405a35378982b2b3c16ef4160282ea1c7f8fbda25422
2025-04-17 07:49:30.814 [DBG] CORE[firo]: returning coins [d140fd7d3f4a0d1520cc0690b136710b283fd32bfd3137b57acfca5ccb3eb057:0]
2025-04-17 07:49:30.821 [INF] CORE: Order c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 canceled. match id = 9933d3d0dfc4024623437e821e9f2194c68887358c3939d17cf319f0c1d4bcd1
2025-04-17 07:49:30.832 [INF] CORE: Order 48903bcae33988bfcc46e25acd4102970d24f6833dc8acff89b8747f67366fe8 canceled. match id = 0b3ce09a1c8bbd05d06f7c8e9aa0b3e46ef07475f02ae0a44edb242c8b0d3a64
2025-04-17 07:49:30.835 [INF] CORE: Order a8db5e7eac5d133a86e4b0ba46898157accd82caccdc3d13ed6bb35edb30b3ba canceled. match id = beecbb210f0466152333af6d33765901b037b61884a75ab411e9e7ca9e8aad12
2025-04-17 07:49:30.862 [DBG] CORE: notify: |POKE| (order) Order canceled - Buy order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5}}}) - Order: c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5
2025-04-17 07:49:30.874 [DBG] CORE: notify: |POKE| (order) Order canceled - Buy order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|18320727da8c65973a81c6f65c6f158a97e63e3d7b60d0758d15587240d6343c}}}) - Order: 18320727da8c65973a81c6f65c6f158a97e63e3d7b60d0758d15587240d6343c
2025-04-17 07:49:30.877 [DBG] CORE: notify: |POKE| (order) Order canceled - Sell order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|ca85496ab036a19b998f3b4535339d2519d4b3dc59747c9ee1c327c0543d77a1}}}) - Order: ca85496ab036a19b998f3b4535339d2519d4b3dc59747c9ee1c327c0543d77a1
2025-04-17 07:49:30.881 [DBG] CORE: notify: |POKE| (order) Order canceled - Sell order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|14d4f6f7134c12facb006b6dbc300f8ba937df0d1c3b5347495816c9a4ed2e19}}}) - Order: 14d4f6f7134c12facb006b6dbc300f8ba937df0d1c3b5347495816c9a4ed2e19
2025-04-17 07:49:30.885 [DBG] CORE: notify: |POKE| (order) Order canceled - Sell order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|30b58df021688fa23afb97d35fa756890fd60f999080bb61fb705f4c9edddd41}}}) - Order: 30b58df021688fa23afb97d35fa756890fd60f999080bb61fb705f4c9edddd41
2025-04-17 07:49:30.896 [DBG] CORE: notify: |POKE| (order) Order canceled - Buy order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|48903bcae33988bfcc46e25acd4102970d24f6833dc8acff89b8747f67366fe8}}}) - Order: 48903bcae33988bfcc46e25acd4102970d24f6833dc8acff89b8747f67366fe8
2025-04-17 07:49:30.901 [DBG] CORE: notify: |POKE| (order) Order canceled - Buy order on firo-usdc.polygon at dex.decred.org:7232 has been canceled ({{{order|a8db5e7eac5d133a86e4b0ba46898157accd82caccdc3d13ed6bb35edb30b3ba}}}) - Order: a8db5e7eac5d133a86e4b0ba46898157accd82caccdc3d13ed6bb35edb30b3ba
2025-04-17 07:49:31.078 [DBG] CORE: Expired bond 8fb3381a54a6fba9e66fe6c657a0fd43de303f21fad323d1b7905c7b34671bc3:0 (dcr) refundable in about 77h0m10s.
2025-04-17 07:49:31.466 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 573.150237ms, blocked for 1.158µs
2025-04-17 07:49:31.483 [INF] CORE: runJob(match) completed in 689.594697ms
2025-04-17 07:49:32.066 [DBG] CORE: notify: |POKE| (order) Order placed - Buying 10.00000000 FIRO, rate = 0.35718 ({{{order|4dca1d4a37d7ab7ea3e9a2b7640fe5ff3e84a87d170668b7c4df8c3717c3aaea}}}) - Order: 4dca1d4a37d7ab7ea3e9a2b7640fe5ff3e84a87d170668b7c4df8c3717c3aaea
2025-04-17 07:49:32.548 [DBG] CORE: notify: |POKE| (order) Order placed - Buying 10.00000000 FIRO, rate = 0.355334 ({{{order|13941de37fa16e15d24c9d176d3d28f5b8e582f2ac70f341044042e15be534b4}}}) - Order: 13941de37fa16e15d24c9d176d3d28f5b8e582f2ac70f341044042e15be534b4
2025-04-17 07:49:33.008 [DBG] CORE: notify: |POKE| (order) Order placed - Buying 10.00000000 FIRO, rate = 0.353488 ({{{order|fe16e77b5209506de1111f98c5b7d955de460c221f470a56b5666718d6de5ddd}}}) - Order: fe16e77b5209506de1111f98c5b7d955de460c221f470a56b5666718d6de5ddd
2025-04-17 07:49:33.009 [DBG] CORE[firo]: Attempting to fund a multi-order for firo, maxFeeRate = 20
2025-04-17 07:49:33.544 [DBG] CORE: notify: |POKE| (order) Order placed - Selling 10.00000000 FIRO, rate = 0.381216 ({{{order|39b50fe4240f7378da693cafcf3efcbce390cf45e7e94271e6fbf097b50d0e98}}}) - Order: 39b50fe4240f7378da693cafcf3efcbce390cf45e7e94271e6fbf097b50d0e98
2025-04-17 07:49:34.166 [DBG] CORE: notify: |POKE| (order) Order placed - Selling 10.00000000 FIRO, rate = 0.383062 ({{{order|93beda517e45da616bcb2e16d0f4bd38a5964ca20ca27cb39e7815fc6404dd9e}}}) - Order: 93beda517e45da616bcb2e16d0f4bd38a5964ca20ca27cb39e7815fc6404dd9e
2025-04-17 07:49:34.639 [DBG] CORE: notify: |POKE| (order) Order placed - Selling 10.00000000 FIRO, rate = 0.384908 ({{{order|d5337b4caf3b188faa5b5cbc23fde9ee4166443d3ad35236b8d1c12d15967b86}}}) - Order: d5337b4caf3b188faa5b5cbc23fde9ee4166443d3ad35236b8d1c12d15967b86
2025-04-17 07:49:45.147 [DBG] CORE: Received preimage request for order 13941de37fa16e15d24c9d176d3d28f5b8e582f2ac70f341044042e15be534b4 with known commitment 0f583e2f34cb639bd402e70678b809b5bc9f90fe280d695190d666f8c6f60f8f
2025-04-17 07:49:45.147 [DBG] CORE: Received preimage request for order 39b50fe4240f7378da693cafcf3efcbce390cf45e7e94271e6fbf097b50d0e98 with known commitment 3404085a95e727fa9ac6c1a9528326dd6b0d656dfff309cc36b049f1c38e0597
2025-04-17 07:49:45.147 [DBG] CORE: Received preimage request for order 4dca1d4a37d7ab7ea3e9a2b7640fe5ff3e84a87d170668b7c4df8c3717c3aaea with known commitment 5214fb684b647dc96118e07b2b5c9964956a15b1bf3479e4a96ff74fa50596b3
2025-04-17 07:49:45.147 [DBG] CORE: Received preimage request for order 93beda517e45da616bcb2e16d0f4bd38a5964ca20ca27cb39e7815fc6404dd9e with known commitment 876c82f500ae7c39e26e1df5e98e2f2f153970f4e2c79fa4a831a01087229895
2025-04-17 07:49:45.147 [DBG] CORE: Received preimage request for order d5337b4caf3b188faa5b5cbc23fde9ee4166443d3ad35236b8d1c12d15967b86 with known commitment 94a8b373cfbbf5f69b8c762bf5da6500fc5ddb1d47f1c0dfeaae8a67251f441a
2025-04-17 07:49:45.147 [DBG] CORE: Received preimage request for order fe16e77b5209506de1111f98c5b7d955de460c221f470a56b5666718d6de5ddd with known commitment c6ce5456b0f7f3d8870ec126bb2861710caf4256b7400bdceb0b90b173505594
2025-04-17 07:49:45.148 [DBG] CORE: async processPreimageRequest for 13941de37fa16e15d24c9d176d3d28f5b8e582f2ac70f341044042e15be534b4 succeeded
2025-04-17 07:49:45.148 [DBG] CORE: async processPreimageRequest for 4dca1d4a37d7ab7ea3e9a2b7640fe5ff3e84a87d170668b7c4df8c3717c3aaea succeeded
2025-04-17 07:49:45.148 [DBG] CORE: async processPreimageRequest for 39b50fe4240f7378da693cafcf3efcbce390cf45e7e94271e6fbf097b50d0e98 succeeded
2025-04-17 07:49:45.149 [DBG] CORE: async processPreimageRequest for fe16e77b5209506de1111f98c5b7d955de460c221f470a56b5666718d6de5ddd succeeded
2025-04-17 07:49:45.149 [DBG] CORE: async processPreimageRequest for 93beda517e45da616bcb2e16d0f4bd38a5964ca20ca27cb39e7815fc6404dd9e succeeded
2025-04-17 07:49:45.149 [DBG] CORE: async processPreimageRequest for d5337b4caf3b188faa5b5cbc23fde9ee4166443d3ad35236b8d1c12d15967b86 succeeded
2025-04-17 07:49:45.491 [INF] CORE: Standing order 4dca1d4a37d7ab7ea3e9a2b7640fe5ff3e84a87d170668b7c4df8c3717c3aaea did not match and is now booked.
2025-04-17 07:49:45.505 [INF] CORE: Standing order 13941de37fa16e15d24c9d176d3d28f5b8e582f2ac70f341044042e15be534b4 did not match and is now booked.
2025-04-17 07:49:45.508 [INF] CORE: Standing order 93beda517e45da616bcb2e16d0f4bd38a5964ca20ca27cb39e7815fc6404dd9e did not match and is now booked.
2025-04-17 07:49:45.512 [INF] CORE: Standing order 39b50fe4240f7378da693cafcf3efcbce390cf45e7e94271e6fbf097b50d0e98 did not match and is now booked.
2025-04-17 07:49:45.517 [INF] CORE: Standing order fe16e77b5209506de1111f98c5b7d955de460c221f470a56b5666718d6de5ddd did not match and is now booked.
2025-04-17 07:49:45.521 [INF] CORE: Standing order d5337b4caf3b188faa5b5cbc23fde9ee4166443d3ad35236b8d1c12d15967b86 did not match and is now booked.
2025-04-17 07:50:14.432 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 572.056642ms, blocked for 906ns
2025-04-17 07:50:14.565 [DBG] CORE: Slow tick: trade 93beda517e45da616bcb2e16d0f4bd38a5964ca20ca27cb39e7815fc6404dd9e processed in 706.098697ms, blocked for 949ns
2025-04-17 07:50:14.565 [DBG] CORE: Slow tick: trade d5337b4caf3b188faa5b5cbc23fde9ee4166443d3ad35236b8d1c12d15967b86 processed in 706.091752ms, blocked for 2.464µs
2025-04-17 07:50:14.565 [DBG] CORE: Slow tick: trade 30b58df021688fa23afb97d35fa756890fd60f999080bb61fb705f4c9edddd41 processed in 705.784932ms, blocked for 1.787µs
2025-04-17 07:50:14.565 [DBG] CORE: Slow tick: trade ca85496ab036a19b998f3b4535339d2519d4b3dc59747c9ee1c327c0543d77a1 processed in 705.667617ms, blocked for 1.423µs
2025-04-17 07:50:14.565 [DBG] CORE: Slow tick: trade 14d4f6f7134c12facb006b6dbc300f8ba937df0d1c3b5347495816c9a4ed2e19 processed in 705.684227ms, blocked for 1.897µs
2025-04-17 07:50:14.565 [DBG] CORE: Slow tick: trade 39b50fe4240f7378da693cafcf3efcbce390cf45e7e94271e6fbf097b50d0e98 processed in 705.917096ms, blocked for 774ns
2025-04-17 07:50:41.065 [DBG] CORE: Retiring inactive order 48903bcae33988bfcc46e25acd4102970d24f6833dc8acff89b8747f67366fe8 in status canceled
2025-04-17 07:50:41.066 [DBG] CORE: Retiring inactive order a8db5e7eac5d133a86e4b0ba46898157accd82caccdc3d13ed6bb35edb30b3ba in status canceled
2025-04-17 07:50:41.066 [DBG] CORE: Retiring inactive order 30b58df021688fa23afb97d35fa756890fd60f999080bb61fb705f4c9edddd41 in status canceled
2025-04-17 07:50:41.066 [DBG] CORE: Retiring inactive order 14d4f6f7134c12facb006b6dbc300f8ba937df0d1c3b5347495816c9a4ed2e19 in status canceled
2025-04-17 07:50:41.066 [DBG] CORE: Retiring inactive order 18320727da8c65973a81c6f65c6f158a97e63e3d7b60d0758d15587240d6343c in status canceled
2025-04-17 07:50:41.066 [DBG] CORE: Retiring inactive order ca85496ab036a19b998f3b4535339d2519d4b3dc59747c9ee1c327c0543d77a1 in status canceled
2025-04-17 07:50:41.641 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 575.115706ms, blocked for 1.002µs
2025-04-17 07:51:19.433 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 578.51591ms, blocked for 1.421µs
2025-04-17 07:51:48.711 [INF] CORE: Match 596ab69a52e622a89501d6a0bac44ca25705b64dca160a628ebd8fa7b960c62b not yet redeemable: current confs = 1, required confs = 2
2025-04-17 07:51:48.897 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 731.199749ms, blocked for 2.607µs
2025-04-17 07:52:25.367 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 830.451659ms, blocked for 1.354µs
2025-04-17 07:52:29.456 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 903.353403ms, blocked for 978ns
2025-04-17 07:52:39.312 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 760.72926ms, blocked for 446ns
2025-04-17 07:53:08.714 [DBG] CORE: Redeemable match 596ab69a52e622a89501d6a0bac44ca25705b64dca160a628ebd8fa7b960c62b for order c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 (Maker)
2025-04-17 07:53:08.934 [INF] CORE: Broadcasted redeem transaction spending 1 contracts for order c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5, paying to 1c7472de859bf9bbd4fc017532ea2e07783bb9db9a42df4778e6bc57bee15fa1:0 (firo)
2025-04-17 07:53:08.944 [INF] CORE: Match 596ab69a52e622a89501d6a0bac44ca25705b64dca160a628ebd8fa7b960c62b complete: buy 100000000 firo
2025-04-17 07:53:08.946 [DBG] CORE: Notifying DEX dex.decred.org:7232 of our firo swap redemption 1c7472de859bf9bbd4fc017532ea2e07783bb9db9a42df4778e6bc57bee15fa1:0 for match 596ab69a52e622a89501d6a0bac44ca25705b64dca160a628ebd8fa7b960c62b
2025-04-17 07:53:08.947 [DBG] CORE: notify: |POKE| (order) Match complete - Redeemed 1.00000000 FIRO on order {{{order|c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5}}} - Order: c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5
2025-04-17 07:53:08.947 [DBG] CORE: Slow tick: trade c5247c0b082b9d67e064659fe1b2464f93bbb99526cf31a25473210d28c56dd5 processed in 777.013803ms, blocked for 427ns
```
</details>

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12177e2]

goroutine 9622 [running]:
decred.org/dcrdex/client/asset/btc.(*BadgerTxDB).GetTx(0x0, {0xc027391380, 0x40})
	/home/dev/dex/dcrdex/client/asset/btc/txdb.go:394 +0x42
decred.org/dcrdex/client/asset/btc.(*ExchangeWalletElectrum).WalletTransaction(0xc000890a50, {0x1b97860?, 0xc0008a2000?}, {0xc0011c6370, 0x48})
	/home/dev/dex/dcrdex/client/asset/btc/electrum.go:550 +0x13c
decred.org/dcrdex/client/core.(*xcWallet).WalletTransaction(0xc0008a2000, {0x265ce30, 0xc000346aa0}, {0xc0011c6370, 0x48})
	/home/dev/dex/dcrdex/client/core/wallet.go:583 +0x8e
decred.org/dcrdex/client/core.(*Core).WalletTransaction(0xc0005cac00, 0x88, {0xc0011c6370, 0x48})
	/home/dev/dex/dcrdex/client/core/core.go:5817 +0xf0
decred.org/dcrdex/client/mm.(*pendingDEXOrder).updateState.func1(...)
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2704
decred.org/dcrdex/client/mm.(*pendingDEXOrder).updateState(0xc003029900, 0xc00164a180, 0xc002379d90, 0x1084a, 0x56208)
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2714 +0x4f5
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).handleDEXOrderUpdate(0xc0013ce000, 0xc00164a180)
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2741 +0x199
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).handleDEXNotification(0xc0013ce000, {0x2668fa0?, 0xc002399500?})
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:2815 +0x545
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect.func3()
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:3427 +0xc5
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect in goroutine 9022
	/home/dev/dex/dcrdex/client/mm/exchange_adaptor.go:3417 +0x35b
```

--- a/client/asset/btc/btc.go
+++ b/client/asset/btc/btc.go
@@ -811,6 +811,10 @@ type baseWallet struct {
 	feeCache          *feeRateCache
 	decodeAddr        dexbtc.AddressDecoder
 	walletDir         string
+	// noListTxHistory is true for assets that cannot call the
+	// ListTransactionSinceBlock method. This is true for Firo
+	// electrum as of electrum 4.1.5.3.
+	noListTxHistory bool
 
 	deserializeTx func([]byte) (*wire.MsgTx, error)
 	serializeTx   func(*wire.MsgTx) ([]byte, error)
@@ -5899,6 +5903,10 @@ func (btc *baseWallet) idUnknownTx(tx *ListTransactionsResult) (*asset.WalletTra
 func (btc *baseWallet) addUnknownTransactionsToHistory(tip uint64) {
 	txHistoryDB := btc.txDB()
 	if txHistoryDB == nil {
+		return
+	}
+
+	if btc.noListTxHistory {
 		return
 	}
 

--- a/client/asset/btc/electrum.go
+++ b/client/asset/btc/electrum.go
@@ -76,11 +76,16 @@ func ElectrumWallet(cfg *BTCCloneCFG) (*ExchangeWalletElectrum, error) {
 		findRedemptionQueue: make(map[OutPoint]*FindRedemptionReq),
 		minElectrumVersion:  cfg.MinElectrumVersion,
 	}
+
 	// In (*baseWallet).feeRate, use ExchangeWalletElectrum's walletFeeRate
 	// override for localFeeRate. No externalFeeRate is required but will be
 	// used if eew.walletFeeRate returned an error and an externalFeeRate is
 	// enabled.
 	btc.localFeeRate = eew.walletFeeRate
+
+	// Firo electrum does not have "onchain_history" method as of firo
+	// electrum 4.1.5.3, find an alternative.
+	btc.noListTxHistory = cfg.Symbol == "firo"
 
 	return eew, nil
 }
@@ -150,22 +155,16 @@ func (btc *ExchangeWalletElectrum) Connect(ctx context.Context) (*sync.WaitGroup
 		btc.ew.wallet.SetIncludeIgnoreWarnings(true)
 	}
 
-	// TODO: Firo electrum does not have "onchain_history" method as of firo
-	// electrum 4.1.5.3, find an alternative.
-	hasOnchainHistory := btc.symbol != "firo"
-
-	if hasOnchainHistory {
-		dbWG, err := btc.startTxHistoryDB(ctx)
-		if err != nil {
-			return nil, err
-		}
-
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			dbWG.Wait()
-		}()
+	dbWG, err := btc.startTxHistoryDB(ctx)
+	if err != nil {
+		return nil, err
 	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		dbWG.Wait()
+	}()
 
 	wg.Add(1)
 	go func() {
@@ -179,16 +178,14 @@ func (btc *ExchangeWalletElectrum) Connect(ctx context.Context) (*sync.WaitGroup
 		btc.monitorPeers(ctx)
 	}()
 
-	if hasOnchainHistory {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			btc.tipMtx.RLock()
-			tip := btc.currentTip
-			btc.tipMtx.RUnlock()
-			go btc.syncTxHistory(uint64(tip.Height))
-		}()
-	}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		btc.tipMtx.RLock()
+		tip := btc.currentTip
+		btc.tipMtx.RUnlock()
+		go btc.syncTxHistory(uint64(tip.Height))
+	}()
 
 	return wg, nil
 }
@@ -547,6 +544,9 @@ func (btc *ExchangeWalletElectrum) WalletTransaction(ctx context.Context, txID s
 	}
 
 	txHistoryDB := btc.txDB()
+	if txHistoryDB == nil {
+		return nil, fmt.Errorf("tx database not initialized")
+	}
 	tx, err := txHistoryDB.GetTx(txID)
 	if err != nil && !errors.Is(err, asset.CoinNotFoundError) {
 		return nil, err

